rmf_schedule_visualizer/src/rmf_schedule_visualizer/TrajectoryServer.cpp

```diff
@@ -377,19 +350,129 @@ std::string Server::parse_trajectories(
   return response;
 }
 
-Server::~Server()
+//==============================================================================
+std::shared_ptr<TrajectoryServer> TrajectoryServer::make(
```

Comment:
Consider moving some of this inside the pimpl object to shorten and simplify this function.